### PR TITLE
Fix compilation with latest roslyn

### PR DESCRIPTION
**Pull request by am11**
Related PR in runtime: https://github.com/dotnet/runtime/pull/109467

To unblock https://github.com/dotnet/sdk/pull/43015, errors like these:
> /vmr/src/aspnetcore/src/DataProtection/DataProtection/src/Internal/ContainerUtils.cs(106,31): error CS0023: Operator '.' cannot be applied to operand of type 'void' [/vmr/src/aspnetcore/src/DataProtection/DataProtection/src/Microsoft.AspNetCore.DataProtection.csproj::TargetFramework=netstandard2.0]

### CODE REVIEW

**Comment by am11** [comment]
@ViktorHofer, I wasn't able to update packages to 4.13.0-1.24510.9 because it's failing with `[DiagnosticAnalyzer(LanguageNames.CSharp)]`:

> /workspaces/aspnetcore/src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteHandlers/RouteHandlerAnalyzer.cs(18,2): error RS1038: This compiler extension should not be implemented in an assembly containing a reference to Microsoft.CodeAnalysis.Workspaces. The Microsoft.CodeAnalysis.Workspaces assembly is not provided during command line compilation scenarios, so references to it could cause the compiler extension to behave unpredictably. (https://github.com/dotnet/roslyn-analyzers/blob/main/docs/rules/RS1038.md)

so these are the code changes only, based on search/replace. Could you try applying it as a patch to source-build? https://github.com/dotnet/aspnetcore/pull/58769.patch

File: src/Components/test/testassets/BasicTestApp/EventDuringBatchRendering.razor [hunk 1]
```diff
@@ -45,7 +45,7 @@
     @foreach (var item in itemsList)
     {
         <div @key="item">
-            <input @oninput="@(() => itemsList.Reverse())"
+            <input @oninput="@(() => Enumerable.Reverse(itemsList))"
```
Comment (am11) [inline src/Components/test/testassets/BasicTestApp/EventDuringBatchRendering.razor:48]:
```suggestion
            <input @oninput="@(() => itemsList.Reverse())"
```

**Comment by am11** [comment]
The VMR legs failures in linked issue are related: https://github.com/dotnet/sdk/runs/32426310644

To make sure that we have covered all the potential issues we should update Versions.props and Version.Details.xml as part of this PR with `4.13.0-1.24510.9` (currently used by runtime and sdk), but when I tried there were some dependency issues which I'm not sure how to resolve https://github.com/dotnet/aspnetcore/pull/58769#issuecomment-2453115466.

**Comment by am11** [comment]
The VMR legs failures in linked issue are related: https://github.com/dotnet/sdk/runs/32426310644

To make sure that we have covered all the potential issues we should update Versions.props and Version.Details.xml as part of this PR with `4.13.0-1.24510.9` (currently used by runtime and sdk), but when I tried there were some dependency issues which I'm not sure how to resolve https://github.com/dotnet/aspnetcore/pull/58769#issuecomment-2453115466.

**Comment by am11** [comment]
> is this actually failing currently? Expected to fail if we update the compiler? Or ...?

The VMR legs failures in linked issue are related: https://github.com/dotnet/sdk/runs/32426310644

To make sure that we have covered all the potential issues we should update Versions.props and Version.Details.xml as part of this PR with `4.13.0-1.24510.9` (currently used by runtime and sdk), but when I tried there were some dependency issues which I'm not sure how to resolve https://github.com/dotnet/aspnetcore/pull/58769#issuecomment-2453115466.

**Comment by mgravell** [comment]
Relevant context is in [this comment](https://github.com/dotnet/runtime/pull/108789#discussion_r1796995408) - basically, new extension method overloads conflicting, specifically `Reverse`.

File: src/Framework/App.Runtime/src/CompatibilitySuppressions.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
+<!-- https://learn.microsoft.com/dotnet/fundamentals/package-validation/diagnostic-ids -->
```
Comment (mgravell) [inline src/Framework/App.Runtime/src/CompatibilitySuppressions.xml:2]: 👍

**Review by mgravell — APPROVED** [review]
The changes LGTM; can I check, however; is this actually failing currently? Expected to fail if we update the compiler? Or ...?

**Comment by ViktorHofer** [comment]
Yes, this is causing failures in https://github.com/dotnet/sdk/pull/43015 which updates the SDK and builds the aspnetcore repo as part of the VMR. Let's get this merged to unblock dependency flow.

**Pull request merged**